diff --git a/gdb/gdbsupport/pathstuff.c b/gdb/gdbsupport/pathstuff.c
index aa51e8f36e..3412eb23cb 100644
--- a/gdb/gdbsupport/pathstuff.c
+++ b/gdb/gdbsupport/pathstuff.c
@@ -262,7 +262,7 @@ get_standard_temp_dir ()
   if (tmp != nullptr)
     return tmp;
 
-  return "/tmp";
+  return "@TERMUX_PREFIX@/tmp";
 #endif
 }
 
@@ -273,7 +273,7 @@ get_shell ()
 {
   const char *ret = getenv ("SHELL");
   if (ret == NULL)
-    ret = "/bin/sh";
+    ret = "@TERMUX_PREFIX@/bin/bash";
 
   return ret;
 }
