diff --git a/gnulib/import/string.in.h b/gnulib/import/string.in.h
index 9b8ced24cd..22d2dc1606 100644
--- a/gnulib/import/string.in.h
+++ b/gnulib/import/string.in.h
@@ -200,9 +200,8 @@ _GL_WARN_ON_USE (memmem, "memmem is unportable and often quadratic - "
 #if @GNULIB_MEMPCPY@
 # if ! @HAVE_MEMPCPY@
 _GL_FUNCDECL_SYS (mempcpy, void *,
-                  (void *restrict __dest, void const *restrict __src,
-                   size_t __n)
-                  _GL_ARG_NONNULL ((1, 2)));
+                  (void * __dest, void const * __src,
+                   size_t __n));
 # endif
 _GL_CXXALIAS_SYS (mempcpy, void *,
                   (void *restrict __dest, void const *restrict __src,
