--- ./src/makemake.in	2010-07-07 15:15:35.000000000 +0000
+++ ../makemake.in	2017-05-09 20:02:09.707937833 +0000
@@ -2855,7 +2855,7 @@
 # and 1500KW. To reduce GCs, we spend 30% more than this. 8MB is not a big deal.
 MEMOPT="-m 2MW"
 
-if [ $CROSS = false ] ; then
+# if [ $CROSS = false ] ; then
 
   if [ "$HSYS" = win32gcc ]; then
     # work around an msys "feature": ":" after "/" is converted to a ";"
@@ -2880,9 +2880,9 @@
   echotab "\$(MV) lispimag.mem interpreted.mem"
   echol
 
-fi
+# fi
 
-if [ $CROSS = false ] ; then
+# if [ $CROSS = false ] ; then
 
   for f in $LPARTS ; do
     if [ $f = "compiler" ]; then
@@ -2919,7 +2919,7 @@
   echotab '$(RUN) -x "(and (load \"init.fas\") (ext::saveinitmem) (ext::exit)) (ext::exit t)"'
   echol
 
-fi
+# fi
 
 echol
 
