--- ./dwarfdump/Makefile.in	2017-08-14 20:03:41.000000000 +0000
+++ ../dwarfdump-Makefile.in	2018-03-08 20:18:52.864300481 +0000
@@ -36,7 +36,7 @@
 DIRINC =        $(srcdir)/../libdwarf
 LIBS =		@LIBS@ -L../libdwarf -ldwarf -lelf $(dwfzlib)
 INCLUDES =	-I. -I$(srcdir) -I../libdwarf -I$(srcdir)/../libdwarf
-CFLAGS =	$(PREINCS) @CPPFLAGS@ @CFLAGS@ $(INCLUDES) $(dwfwall) $(dwfsanitize) -DCONFPREFIX=${libdir} $(POSTINCS)
+CFLAGS =	$(PREINCS) $(INCLUDES) @CPPFLAGS@ @CFLAGS@ $(dwfwall) $(dwfsanitize) -DCONFPREFIX=${libdir} $(POSTINCS)
 LDFLAGS =	$(PRELIBS) @LDFLAGS@  $(LIBS) $(dwfsanitize) $(POSTLIBS)
 #VG =            valgrind --leak-check=full  --show-leak-kinds=all
 VG =
@@ -123,45 +123,6 @@
 naming.o:  $(srcdir)/naming.c $(srcdir)/naming.h
 	$(CC) $(CFLAGS) -c $(srcdir)/naming.c
 
-# We need this as naming.o has external references we cannot have
-# in the tree builds.
-trivial_naming.o: $(srcdir)/naming.c
-	$(CC) $(CFLAGS) -DTRIVIAL_NAMING  -c $(srcdir)/naming.c -o trivial_naming.o 
-
-tag_tree_build:  $(srcdir)/tag_tree.c $(DIRINC)/dwarf.h $(HEADERS) tag_common.o makename.o common.o trivial_naming.o dwgetopt.o dwarf_tsearchbal.o
-	LD_LIBRARY_PATH=$$LD_LIBRARY_PATH:$(LIBDWARF_PATH) $(CC) $(CFLAGS) $(srcdir)/tag_tree.c tag_common.o common.o makename.o  dwarf_tsearchbal.o trivial_naming.o dwgetopt.o  $(LDFLAGS) -o tag_tree_build 
-
-tag_attr_build:  $(srcdir)/tag_attr.c $(DIRINC)/dwarf.h $(HEADERS) tag_common.o makename.o  common.o trivial_naming.o dwgetopt.o dwarf_tsearchbal.o
-	$(CC) $(CFLAGS) $(srcdir)/tag_attr.c tag_common.o common.o makename.o trivial_naming.o  dwarf_tsearchbal.o dwgetopt.o $(LDFLAGS) -o tag_attr_build 
-
-# Plain GNU C dash E does not work on a .list, 
-# so copy to a .c name to run
-# the following four table creations.
-tmp-tt-table.c: $(srcdir)/tag_tree.list tag_tree_build
-	-rm -f  tmp-t1.c
-	cp $(srcdir)/tag_tree.list tmp-t1.c
-	$(CC) $(CFLAGS) -E tmp-t1.c  > ./tmp-tag-tree-build1.tmp
-	LD_LIBRARY_PATH=$$LD_LIBRARY_PATH:$(LIBDWARF_PATH) ./tag_tree_build -s  -i tmp-tag-tree-build1.tmp  -o tmp-tt-table.c
-
-tmp-tt-ext-table.c: $(srcdir)/tag_tree_ext.list tag_tree_build
-	-rm -f tmp-t4.c
-	cp $(srcdir)/tag_tree_ext.list tmp-t4.c
-	$(CC) $(CFLAGS) -E tmp-t4.c  > ./tmp-tag-tree-build4.tmp
-	LD_LIBRARY_PATH=$$LD_LIBRARY_PATH:$(LIBDWARF_PATH) ./tag_tree_build -e  -i tmp-tag-tree-build4.tmp -o tmp-tt-ext-table.c
-
-tmp-ta-table.c: $(srcdir)/tag_attr.list tag_attr_build
-	-rm -f tmp-t2.c
-	cp $(srcdir)/tag_attr.list tmp-t2.c
-	$(CC) $(CFLAGS)  -E tmp-t2.c  > ./tmp-tag-attr-build2.tmp
-	LD_LIBRARY_PATH=$$LD_LIBRARY_PATH:$(LIBDWARF_PATH) ./tag_attr_build -s -i tmp-tag-attr-build2.tmp -o tmp-ta-table.c
-
-tmp-ta-ext-table.c: $(srcdir)/tag_attr_ext.list tag_attr_build
-	-rm -f tmp-t3.c
-	cp $(srcdir)/tag_attr_ext.list tmp-t3.c
-	$(CC) $(CFLAGS)  -E tmp-t3.c  > ./tmp-tag-attr-build3.tmp
-	LD_LIBRARY_PATH=$$LD_LIBRARY_PATH:$(LIBDWARF_PATH) ./tag_attr_build -e -i tmp-tag-attr-build3.tmp -o tmp-ta-ext-table.c
-
-
 test: esb.o dwgetopt.o 
 	echo "ESB test"
 	$(CC) $(CFLAGS) -o test $(srcdir)/testesb.c esb.o
@@ -231,7 +192,6 @@
 	rm -f tag_tree_build
 	rm -f tag_attr_build
 	-rm -f _*.c _*.h
-	-rm -f tmp-*.c tmp-*.h tmp-*.tmp
 	rm -f gennames
 	rm -f dwarf_names_new.c
 	rm -f dwarf_names_new.h
