Patch assimilated from archlinux

diff -u -r ../xetexdir.orig/pdfimage.cpp ./texk/web2c/xetexdir/pdfimage.cpp
--- ../xetexdir.orig/pdfimage.cpp	2019-03-02 07:44:23.079981805 +0100
+++ ./texk/web2c/xetexdir/pdfimage.cpp	2019-03-02 07:50:05.639979391 +0100
@@ -82,19 +82,19 @@
 	switch (pdf_box) {
 		default:
 		case pdfbox_crop:
-			r = page->getCropBox();
+			r = (PDFRectangle *)page->getCropBox();
 			break;
 		case pdfbox_media:
-			r = page->getMediaBox();
+			r = (PDFRectangle *)page->getMediaBox();
 			break;
 		case pdfbox_bleed:
-			r = page->getBleedBox();
+			r = (PDFRectangle *)page->getBleedBox();
 			break;
 		case pdfbox_trim:
-			r = page->getTrimBox();
+			r = (PDFRectangle *)page->getTrimBox();
 			break;
 		case pdfbox_art:
-			r = page->getArtBox();
+			r = (PDFRectangle *)page->getArtBox();
 			break;
 	}
 
