--- ./etc/config.sh/mpi	2017-06-30 08:50:14.000000000 +0000
+++ ../mpi	2017-08-30 13:31:49.443012788 +0000
@@ -45,9 +45,7 @@
         unset OPAL_PREFIX
     fi
 
-    # Bit of a hack: strip off 'lib' and hope this is the prefix for openmpi
-    # include files and libraries.
-    libDir=$(mpicc --showme:link | sed -e 's/.*-L\([^ ]*\).*/\1/')
+    libDir=@TERMUX_PREFIX@/lib
 
     export MPI_ARCH_PATH="${libDir%/*}"
     _foamAddLib     $libDir
