--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/c	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,16 @@
+SUFFIXES += .c
+
+cWARN        = -Wall
+
+cc          = $(WM_CC)
+
+include $(DEFAULT_RULES)/c$(WM_COMPILE_OPTION)
+
+cFLAGS      = $(GFLAGS) $(cWARN) $(cOPT) $(cDBUG) $(LIB_HEADER_DIRS) -fPIC
+
+ctoo        = $(WM_SCHEDULER) $(cc) $(cFLAGS) -c $< -o $@
+
+LINK_LIBS   = $(cDBUG) -L@TERMUX_PREFIX@/lib
+
+LINKLIBSO   = $(cc) -shared
+LINKEXE     = $(cc) -Xlinker -z -Xlinker nodefs
--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/c++	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,24 @@
+SUFFIXES += .C
+
+c++WARN     = -Wall -Wextra -Wold-style-cast -Wnon-virtual-dtor -Wno-unused-parameter -Wno-invalid-offsetof
+
+# Suppress some warnings for flex++ and CGAL
+c++LESSWARN = -Wno-old-style-cast -Wno-unused-local-typedefs -Wno-array-bounds
+
+CC          = $(WM_CXX) -std=c++11
+
+include $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
+
+ptFLAGS     = -DNoRepository -ftemplate-depth-100
+
+c++FLAGS    = $(GFLAGS) $(c++WARN) $(c++OPT) $(c++DBUG) $(ptFLAGS) $(LIB_HEADER_DIRS) -fPIC
+
+Ctoo        = $(WM_SCHEDULER) $(CC) $(c++FLAGS) -c $< -o $@
+cxxtoo      = $(Ctoo)
+cctoo       = $(Ctoo)
+cpptoo      = $(Ctoo)
+
+LINK_LIBS   = $(c++DBUG) -L@TERMUX_PREFIX@/lib
+
+LINKLIBSO   = $(CC) $(c++FLAGS) -shared -Xlinker
+LINKEXE     = $(CC) $(c++FLAGS) -Xlinker
--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/c++Debug	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,2 @@
+c++DBUG    = -ggdb3 -DFULLDEBUG
+c++OPT      = -O0 -fdefault-inline
--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/c++Opt	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,4 @@
+c++DBUG     =
+c++OPT      = -O3 -floop-optimize -falign-loops -falign-labels -falign-functions -falign-jumps
+
+ROUNDING_MATH = -frounding-math
--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/c++Prof	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,2 @@
+c++DBUG    = -pg
+c++OPT     = -O2
--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/cDebug	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,2 @@
+cDBUG       = -ggdb -DFULLDEBUG
+cOPT        = -O1 -fdefault-inline -finline-functions
--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/cOpt	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,2 @@
+cDBUG       =
+cOPT        = -O3 -floop-optimize -falign-loops -falign-labels -falign-functions -falign-jumps
--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/cProf	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,2 @@
+cDBUG       = -pg
+cOPT        = -O2
--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/general	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,9 @@
+CPP        = cpp -traditional-cpp $(GFLAGS)
+LD         ?= ld
+
+PROJECT_LIBS = -l$(WM_PROJECT) -ldl
+
+include $(GENERAL_RULES)/standard
+
+include $(DEFAULT_RULES)/c
+include $(DEFAULT_RULES)/c++
--- /dev/null	2017-08-30 11:30:49.966075856 +0000
+++ ./wmake/rules/linux64arm-linux-androideabi-gcc/mplibHPMPI	2017-08-30 14:14:55.455582341 +0000
@@ -0,0 +1,3 @@
+PFLAGS     =
+PINC       = -isystem $(MPI_ARCH_PATH)/include -D_MPICC_H
+PLIBS      = -L$(MPI_ARCH_PATH)/lib/linux_ia32 -lmpi
