--- ./src/OSspecific/POSIX/clockTime/clockTime.C	2017-06-30 08:50:14.000000000 +0000
+++ ../clockTime.C	2017-11-23 18:35:12.481633113 +0000
@@ -23,8 +23,8 @@
 
 \*---------------------------------------------------------------------------*/
 
+#include <time.h>
 #include "clockTime.H"
-#include <sys/time.h>
 
 // * * * * * * * * * * * * * Private Member Functions  * * * * * * * * * * * //
 
--- ./src/OSspecific/POSIX/clockTime/clockTime.H	2017-06-30 08:50:14.000000000 +0000
+++ ../clockTime.H	2017-11-23 18:33:37.345638886 +0000
@@ -37,7 +37,7 @@
 #define clockTime_H
 
 #include <sys/types.h>
-
+#include <linux/time.h>
 // * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * //
 
 namespace Foam
