--- ./src/OSspecific/POSIX/POSIX.C	2017-06-30 08:50:14.000000000 +0000
+++ ../POSIX.C	2017-09-13 13:15:23.612141642 +0000
@@ -1107,7 +1109,7 @@
     // Fill sockaddr_in structure with dest address and port
     memset(reinterpret_cast<char *>(&destAddr), '\0', sizeof(destAddr));
     destAddr.sin_family = AF_INET;
-    destAddr.sin_port = htons(ushort(destPort));
+    destAddr.sin_port = htons(u_short(destPort));
     destAddr.sin_addr.s_addr = addr;
 
 
