--- ./wmake/rules/General/general	2017-06-30 08:50:14.000000000 +0000
+++ ../general	2017-08-30 17:35:25.167100785 +0000
@@ -1,17 +1,17 @@
 #-------------------------------*- makefile -*---------------------------------
 WM_VERSION = OPENFOAM_PLUS=1706
 
-AR         = ar
-ARFLAGS    = cr
-RANLIB     = ranlib
-CPP        = cpp
-LD         = ld
+#AR         = aarch64-linux-android-ar
+#ARFLAGS    = aarch64-linux-android-cr
+#RANLIB     = aarch64-linux-android-ranlib
+#CPP        = aarch64-linux-android-cpp
+#LD         = aarch64-linux-android-ld
 
 GFLAGS     = -D$(WM_VERSION) -D$(WM_ARCH) -DWM_ARCH_OPTION=$(WM_ARCH_OPTION) \
              -DWM_$(WM_PRECISION_OPTION) -DWM_LABEL_SIZE=$(WM_LABEL_SIZE)
-GINC       =
+GINC       = -I@TERMUX_PREFIX@/include
 GLIBS      = -lm
-GLIB_LIBS  =
+GLIB_LIBS  = -L@TERMUX_PREFIX@/lib
 
 
 COMPILER_TYPE   = $(shell echo $(WM_COMPILER) | tr -d [:digit:])
